src/utils/theme.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import _ from 'lodash/fp'
+
+let ThemeContext = React.createContext()
+
+export let ThemeProvider = ({ children, theme }) => (
+  <ThemeContext.Provider value={theme}>{children}</ThemeContext.Provider>
+)
+
+export let mergeNestedTheme = (key = '', theme) =>
+  _.flow(
```

Comment:
Let’s only merge if there are any overrides (especially since it’s less common that there will be any)